whelk-core/src/main/groovy/whelk/JsonLdValidator.groovy

```diff
@@ -41,9 +59,26 @@ class JsonLdValidator {
     }
 
     List<Error> validate(Map map) {
-        def validation = new Validation(Validation.Scope.NESTED_GRAPH)
-        doValidate(map, validation)
-        return validation.errors
+        return validate(map, null)
+    }
+
+    List<Error> validate(Map map, String collection) {
+        def scopes = []
+
+        // Scopes that all things should have
+        scopes.addAll(scopesPerLegacyCollection['all'])
+
+        // Scopes that only some types of things should have (for the moment!)
+        if (collection && collection in scopesPerLegacyCollection) {
+            scopes.addAll(scopesPerLegacyCollection[collection])
+        }
+
+        return scopes.inject([], { errors, value ->
```

Comment:
This is fine! But I had to go look it up, and it sort of hurt my head ;(
There are so many annoying ways to be clever in groovy.

Reply:
Was actually thinking that too. I'll make it more readable in the next iteration :)